ophyd_basler/basler_camera.py

```diff
@@ -78,49 +79,58 @@ def __init__(self, *args, root_dir="/tmp/basler", cam_num=0, pixel_format="Mono8
             print("Pixel format                :", self.active_format.get())
             print("Camera min. pixel level     :", self.pixel_level_min.get())
             print("Camera max. pixel level     :", self.pixel_level_max.get())
-            print("Grab timeout                :", self.grab_timeout, "ms")
+            print("Grab timeout                :", self.grab_timeout.get(), "ms")
             print("Trigger mode                :", trigger_mode)
             print("GigE transport payload size : " + "{:,}".format(self.payload_size.get()) + " bytes")
 
-    def grab_images(self):
+    def grab_image(self):
 
-        self.camera_object.StartGrabbingMax(self.exposure_frames.get())
-        images = []
-
-        while self.camera_object.IsGrabbing():
-            grab_result = self.camera_object.RetrieveResult(
-                self.grab_timeout, pylon.TimeoutHandling_ThrowException
+        # Exposure time can't be less than self.camera_object.ExposureTime.Min.
+        # We use seconds for ophyd, and microseconds for pylon:
+        min_exposure_us = self.camera_object.ExposureTimeAbs.Min
+        if min_exposure_us > 1e6 * self.exposure_time.get():
+            self.exposure_time.put(1e-6 * min_exposure_us)
+            warnings.warn(
+                f"Desired exposure time ({1e6 * self.exposure_time.get()} us) is less than "
+                f"the minimum exposure time ({min_exposure_us} us). Proceeding with minimum exposure time."
             )
```

Comment:
I don't think we need to do it on every `grab_image`/`trigger` call. I'd recommend moving it to the `stage` method so that it can be checked only once. We won't be changing exposure time during one acquisition run.